modules/class-simplechart-save.php

```diff
@@ -53,10 +48,71 @@ function save_post_action( $post_id ) {
 
 		remove_action( 'save_post_simplechart', array( $this, 'save_post_action' ), 10, 1 );
 		$post = get_post( $post_id );
-		$this->do_save_post( $post );
+		$this->_do_save_post( $post );
 		add_action( 'save_post_simplechart', array( $this, 'save_post_action' ), 10, 1 );
 	}
 
+	protected function _do_save_post( $post ) {
+		// verify nonce
+		if ( empty( $_POST['simplechart-nonce'] ) || ! wp_verify_nonce( sanitize_text_field( wp_unslash( $_POST['simplechart-nonce'] ) ), 'simplechart_save' ) ) {
+			return;
+		}
+
+		// delete featured image if post is NOT published but has a featured image
+		if ( 'publish' !== get_post_status( $post->ID ) && has_post_thumbnail( $post->ID ) ) {
+			wp_delete_attachment( get_post_thumbnail_id( $post->ID ), true );
+		}
+
+		// handle base64 image string if provided
+		if ( ! empty( $_POST['save-previewImg'] ) ) {
+			$this->_save_chart_image( $post, sanitize_text_field( wp_unslash( $_POST['save-previewImg'] ) ), $this->_default_img_type );
+		}
+
+		// handle raw CSV multiline data
+		if ( ! empty( $_POST['save-rawData'] ) ) {
+			add_filter( 'sanitize_text_field', array( $this, 'sanitize_raw_data' ), 99, 2 );
```

Comment:
@bcampeau Here's how I ended up handling `sanitize_text_field()` for multiline input